### PR TITLE
User is able to edit a tag

### LINKED ISSUE #32: Edit a Tag
As an admin I would like to be able to modify a Tag so that I can rephrase the name if I think of something more appropriate.

**Given** the user is viewing the Tag list  
**When** they select the option to edit an Tag  
**Then** the user should be directed to a form and given the ability to change the Tag's name  

**Given** the user is finished updating the Tag information  
**When** they click the `Save` button  
**Then** the updated Tag should be saved to the database  
**And** the user should be redirected to the new Tag List page  

**Given** the user has decided not to edit the Tag  
**When** they click the `Cancel` button  
**Then** the user should be redirected back to the list page  

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by user1hunter**
#### Changes Made
1. Added file `TagEdit.js` to `components` directory.
2. Modified file `tagManager` to include `EditTag` functionality.
3.  Modified file `TagList` to include `Edit` button
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout HP-editTag
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.
    > Login to application
    > Click on `Tag` navbar link
    > Click on `Edit` button and the user should be redirected to an edit page
    > Upon changing the tag name, user should be redirected to the tags page
    > The tag edited should have it's name updated to what you changed it to
4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

Closes #32

### CODE REVIEW

**Review by nathanelmer — APPROVED** [review]
This looks incredible. I'm crying

File: Tabloid/client/src/components/ApplicationViews.js [hunk 1]
```diff
@@ -49,6 +50,10 @@ export default function ApplicationViews({ isLoggedIn }) {
           {isLoggedIn ? <TagForm /> : <Redirect to="/login" />}
         </Route>
 
+        <Route path="/tags/edit/:id" >
```
Comment (heymonicakay) [inline Tabloid/client/src/components/ApplicationViews.js]: should the id in the route path be an integer? if so, please add some regex to ensure this route path is only matched when it is an integer.

File: Tabloid/client/src/components/TagEdit.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  return (  
+    <Form>
+    <FormGroup>
+      <Label for="name">Name</Label>
+      <Input
+        type="text"
+        name="name"
+        id="name"
+        placeholder={tag.name}
+        value={tag.name}
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagEdit.js]: This value will be undefined on the first page render. Please make sure this is a controlled component by either setting a name property with the value of an empty string on your initial tag object in state, or including logic here to set the value to an empty string when tag.name is undefined.

File: Tabloid/client/src/modules/tagManager.js [hunk 3]
```diff
@@ -13,12 +13,29 @@ export const getAllTags = () => {
...
+      return fetch(`${_apiUrl}/${id}`, {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }).then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          throw new Error("Unknown Error Getting Tags");
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/tagManager.js]: Is this an error getting tags or a single tag?

File: Tabloid/client/src/components/TagList.js [hunk 4]
```diff
@@ -25,7 +26,9 @@ export const TagList = () => {
       <div className="container">
         <div className="row justify-content-center">
           {tags.map((tag) => (
-            <Tag tag={tag} key={tag.id} />
+            <ListGroupItem><Tag tag={tag} key={tag.id} />
```
Comment (heymonicakay) [inline Tabloid/client/src/components/TagList.js]: Is this producing a key error?

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments

File: Tabloid/client/src/components/ApplicationViews.js (hunk 1 above)
Comment (user1hunter) [reply Tabloid/client/src/components/ApplicationViews.js]: Fixed

File: Tabloid/client/src/modules/tagManager.js (hunk 3 above)
Comment (user1hunter) [reply Tabloid/client/src/modules/tagManager.js]: Fixed

File: Tabloid/client/src/components/TagList.js (hunk 4 above)
Comment (user1hunter) [reply Tabloid/client/src/components/TagList.js]: Fixed

File: Tabloid/client/src/components/TagEdit.js (hunk 2 above)
Comment (user1hunter) [reply Tabloid/client/src/components/TagEdit.js]: Fixed

**Review by cjgochanour — APPROVED** [review]
👍

**Review by heymonicakay — APPROVED** [review]

**Pull request merged**